.github/workflows/main.yml

```
@@ -0,0 +1,46 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches:
+      - '*'
+
+jobs:
+
+  Test:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout code
+      - name: Checkout code
+        uses: actions/checkout@v3
```

There is a new version for this one `v4` I think. Worth checking if any of the others have new releases.

It is also recommended from a security perspective to pin these to commit SHAs as they provide immutability, unlike tags which do not. We already do this in a few of our repos. If you then use Dependabot to manage the updates, it is smart enough to automatically update the SHAs, see an example in [this](https://github.com/ral-facilities/inventory-management-system-api/pull/407) IMS API PR.

```suggestion
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```